static/app/views/releases/thresholdsList/thresholdGroupTable.tsx

```diff
@@ -0,0 +1,105 @@
+import {useMemo} from 'react';
+import styled from '@emotion/styled';
+
+import ProjectBadge from 'sentry/components/idBadge/projectBadge';
+import PanelTable from 'sentry/components/panels/panelTable';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {Project} from 'sentry/types';
+
+import {Threshold} from '../utils/types';
+
+import {ThresholdGroupRows} from './thresholdGroupRows';
+
+type Props = {
+  allEnvironmentNames: string[];
+  isError: boolean;
+  isLoading: boolean;
+  project: Project;
+  refetch: () => void;
+  setTempError: (msg: string) => void;
+  thresholds: Threshold[];
+};
+
+export default function ThresholdGroupTable({
+  allEnvironmentNames,
+  isError,
+  isLoading,
+  project,
+  refetch,
+  setTempError,
+  thresholds,
+}: Props) {
+  const thresholdsByEnv: {[key: string]: Threshold[]} = useMemo(() => {
+    const byEnv = {};
+    thresholds.forEach(threshold => {
+      const env = threshold.environment ? threshold.environment.name : '';
+      if (!byEnv[env]) {
+        byEnv[env] = [];
+      }
+      byEnv[env].push(threshold);
+    });
+    return byEnv;
+  }, [thresholds]);
+
+  const flattenedThresholds: Threshold[] = useMemo(
```

Comment:
why do we need to flatten them? just to get a list of thresholds, sorted by env?

Reply:
hmmm.

Yes - i suppose i could sort this rather than constructing the byEnv list.
👍